numpy/_core/src/umath/string_ufuncs.cpp

```diff
@@ -150,42 +98,51 @@ string_comparison_loop(PyArrayMethod_Context *context,
 }
 
 
-template <ENCODING enc>
-static int
-string_add_loop(PyArrayMethod_Context *context,
-                char *const data[], npy_intp const dimensions[],
-                npy_intp const strides[], NpyAuxData *NPY_UNUSED(auxdata))
-{
-    int elsize1 = context->descriptors[0]->elsize;
-    int elsize2 = context->descriptors[1]->elsize;
-    int outsize = context->descriptors[2]->elsize;
-
-    char *in1 = data[0];
-    char *in2 = data[1];
-    char *out = data[2];
-
-    npy_intp N = dimensions[0];
-
-    while (N--) {
-        Buffer<enc> buf1(in1, elsize1);
-        Buffer<enc> buf2(in2, elsize2);
-        Buffer<enc> outbuf(out, outsize);
-        string_add<enc>(buf1, buf2, outbuf);
-
-        in1 += strides[0];
-        in2 += strides[1];
-        out += strides[2];
+template <ENCODING enc, IMPLEMENTED_UNARY_FUNCTIONS f, typename T>
+struct call_buffer_unary_function {
```

Comment:
This whole struct can go if one follows the same logic as in https://github.com/numpy/numpy/pull/25796/commits/a3aebdb0c7c05a500ea7896c9563372f7cd89879